assets/javascripts/discourse-assign/controllers/group-assigned.js.es6

```diff
@@ -12,35 +14,62 @@ export default Controller.extend({
   filterName: "",
   filter: "",
 
+  // I don't know know the subject enough but we should try to remove this one too
+  // one way to do this would be to maybe pass group to group-assigned-filter
+  // and use didReceiveAttrs, I didn't try, but that might be a path forward
```

Comment:
This is solving the problem

- We have the `group` route, which loads the group info, including the assigned count. We use that to put a counter like <kbd>Assigned (3)</kbd> on the tab

- When you're in the `group-assigned` tab, it is possible to unassign topics. Therefore the count in the tab should change

Reloading the entire `group` model seems wasteful, so we're just updating the assignment count when needed.

We can totally move this to an `afterModel` hook on the `group-assigned` route. But I still don't really like how we are modifying data in a parent route 🤔  It violates the DDAU principle.

Maybe `group-assigned` can fire an action `assignedCountChanged(count)` during `afterModel`, and then `group` can implement that action, and update the assignment count.

Reply:
I might be missing something, but I think https://github.com/discourse/discourse-assign/pull/99/commits/3a77beef427670c50cc309db1d41ca766e02e925 should just work ?

Reply:
When any unassign happens, we need to refresh the `group.assignment_count` which is available in the parent route. we need to update `group.assignment_count` as we use it in the parent route too. so I think this might not work here?

Reply:
Please do me a test case where it doesn't work if possible

Reply:
[here](https://github.com/discourse/discourse-assign/blob/83ff23c98ec8d5edcb3090093c4025db1d618ef0/assets/javascripts/discourse/templates/components/group-assigned-menu-item.hbs#L2) we use `assignment_count` from group model. so, whenever a reassign happens we need to update assignment_count from group model

Reply:
ok indeed I missed the connector, in this case I think just doing this: https://github.com/discourse/discourse-assign/pull/99/commits/725a79eb93268a7b5a1a3d855204d66bcff8bb7b

would be better, it's a little bit hacky but okayish IMO, and still better than an observer

Reply:
![Screenshot 2020-08-14 at 3 05 28 AM](https://user-images.githubusercontent.com/35112518/90189556-07ffc380-dddb-11ea-866f-ce4bc5a978c0.png)
This happens when we unassign any topic, tab count remains same and just side-bar counts changes

Reply:
That's Perfect😍